### PR TITLE
Closes #1209 Passive sync of entity lifecycle state

### LINKED ISSUE #1209: Implement skeletal active-passive sync
In the same way that #1208 deals with replicating cache lifecycle events to the passive entity, this task adds support for communicating existing cache lifecycle states during passive sync, before allowing a newly arriving passive entity to enter standby state.

### PR DESCRIPTION
**Pull request by albinsuresh**
Initial implementation of syncing non-data state of entity(the store manager state and server store state). Syncing the client message tracker state is still WIP.

### CODE REVIEW

**Comment by albinsuresh** [comment]
In the current PR multiple messages are sent to the passive to synchronize the entity state. One for the store manager state and one each for each server store state. I'm thinking of combining them all into a single message as I really don't see a reason why they should be separate.

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheStateServiceImpl.java [hunk 1]
```diff
@@ -109,18 +109,26 @@ public ServerStoreImpl getStore(String name) {
...
   }
 
-  public void validate(ValidateStoreManager message) throws ClusterException {
+  @Override
+  public Map<String, ServerSideConfiguration.Pool> getSharedResourcePools() {
+    Map<String, ServerSideConfiguration.Pool> sharedPools = new HashMap<String, ServerSideConfiguration.Pool>();
+    for (Map.Entry<String, ResourcePageSource> entry : sharedResourcePools.entrySet()) {
+      sharedPools.put(entry.getKey(), entry.getValue().getPool());
+    }
+    return sharedPools;
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheStateServiceImpl.java]: We have moved this to Java 8:
`return sharedResourcePools.entrySet().stream().collect(toMap(Map.Entry::getKey, e -> e.getValue().getPool()))`

**Comment by albinsuresh** [comment]
Unified the life cycle state sync messages and added the client tracking information to the sync message.

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java [hunk 2]
```diff
@@ -291,7 +293,22 @@ public void onEviction(long key) {
 
   @Override
   public void synchronizeKeyToPassive(PassiveSynchronizationChannel<EhcacheEntityMessage> syncChannel, int concurrencyKey) {
-    throw new UnsupportedOperationException("Active/passive is not supported yet");
+    if (concurrencyKey == 1) {
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: Please reference `org.ehcache.clustered.server.ConcurrencyStrategies.DefaultConcurrencyStrategy#DEFAULT_KEY` instead of a hard coded value.

**Comment by albinsuresh** [comment]
Addressed the review comments.

**Comment by AbfrmBlr** [comment]
This is not complete until we decide on the sync strategy. See discussion regarding syncing of lifecycle messages for cache on https://github.com/Terracotta-OSS/terracotta-apis/pull/146

**Comment by albinsuresh** [comment]
test this please

File: clustered/client/src/test/java/org/ehcache/clustered/client/ActivePassiveSyncTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        .build();
+
+    ClusteringService service = new ClusteringServiceFactory().create(configuration);
+    service.start(null);
+    EhcacheClientEntity clientEntity = getEntity(service);
+
+    clusterControl.startOneServer();
+    clusterControl.waitForRunningPassivesInStandby();
+    clusterControl.terminateActive();
+    clusterControl.waitForActive();
```
Comment (AbfrmBlr) [inline clustered/client/src/test/java/org/ehcache/clustered/client/ActivePassiveSyncTest.java]: We should not wait for Active here

File: clustered/server/src/main/java/org/ehcache/clustered/server/state/ClientMessageTracker.java [hunk 4]
```diff
@@ -73,4 +73,11 @@ public boolean isConfigureApplicable(UUID clientId, long timestamp) {
     return true;
   }
 
+  public UUID getConfiguredEntityId() {
```
Comment (AbfrmBlr) [inline clustered/server/src/main/java/org/ehcache/clustered/server/state/ClientMessageTracker.java]: This change is not required because this is not set by Active, only a newly promoted active needs this info.

**Comment by albinsuresh** [comment]
Addressed the review comments.

File: clustered/server/src/main/java/org/ehcache/clustered/server/state/ClientMessageTracker.java [hunk 5]
```diff
@@ -72,5 +72,4 @@ public boolean isConfigureApplicable(UUID clientId, long timestamp) {
     }
     return true;
   }
-
 }
```
Comment (AbfrmBlr) [inline clustered/server/src/main/java/org/ehcache/clustered/server/state/ClientMessageTracker.java]: unrelated change in this file

File: clustered/server/src/main/java/org/ehcache/clustered/server/messages/EhcacheSyncMessageCodec.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityMessage;
+import org.ehcache.clustered.common.internal.store.Util;
+import org.terracotta.entity.MessageCodecException;
+import org.terracotta.entity.SyncMessageCodec;
+
+import java.nio.ByteBuffer;
+
+public class EhcacheSyncMessageCodec implements SyncMessageCodec<EhcacheEntityMessage> {
+
```
Comment (AbfrmBlr) [inline clustered/server/src/main/java/org/ehcache/clustered/server/messages/EhcacheSyncMessageCodec.java]: Can you add a test for this codec ?

**Comment by albinsuresh** [comment]
Addressed the review comments.

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
Good to go with green build
